test/framework/filetools.py

```diff
@@ -3560,6 +3607,126 @@ def test_create_unused_dir(self):
         self.assertEqual(path, os.path.join(self.test_prefix, 'file_0'))
         self.assertExists(path)
 
+    def test_create_unused_dirs(self):
+        """Test create_unused_dirs function."""
+        requested_paths = []
+        requested_paths.append(os.path.join(self.test_prefix, 'folder_a'))
+        requested_paths.append(os.path.join(self.test_prefix, 'folder_b'))
+        paths = ft.create_unused_dirs(requested_paths)
+        self.assertEqual(paths, requested_paths)
+        map(lambda p: self.assertExists(p), paths)
+
+        # Repeat with existing folder(s) should create new ones
+        for i in range(10):
+            requested_paths = []
+            requested_paths.append(os.path.join(self.test_prefix, 'folder_a'))
+            requested_paths.append(os.path.join(self.test_prefix, 'folder_b'))
+            paths = ft.create_unused_dirs(requested_paths)
+            self.assertEqual(paths, list(map(lambda p: p + '_%s' % i, requested_paths)))
+            map(lambda p: self.assertExists(p), paths)
+
+        # Skip suffix if a directory with the suffix already exists
+        existing_idx = 1
+        os.mkdir(os.path.join(self.test_prefix, "existing_idx_a"))
+        os.mkdir(os.path.join(self.test_prefix, "existing_idx_b"))
+        os.mkdir(os.path.join(self.test_prefix, f"existing_idx_b_{existing_idx}"))
+
+        def expected_idx(n):
+            if n >= existing_idx:
+                return n + 1
+            return n
+
+        for i in range(3):
+            requested_paths = []
+            requested_paths.append(os.path.join(self.test_prefix, 'existing_idx_a'))
```

Comment:
Maybe use variables for `os.path.join(self.test_prefix, "existing_idx_a|b")` to make it clearer that those are the same folders?